src/components/ConsoleIO/ConsoleIO.test.tsx

```diff
@@ -0,0 +1,68 @@
+import { describe, it, expect, vi } from 'vitest';
+
+import { render, screen, fireEvent } from '../../test/test-utils';
+
+import { ConsoleIO } from './ConsoleIO';
+
+describe('ConsoleIO', () => {
+  it('renders tabs and sample area initially, then custom area when tab clicked', () => {
+    render(<ConsoleIO sampleValue="s" customValue="c" data-testid="consoleio-root" />);
+
+    const root = screen.getByTestId('consoleio-root');
+    expect(root).toBeInTheDocument();
+
+    const sampleTab = screen.getByTestId('tab-sample');
+    const customTab = screen.getByTestId('tab-custom');
+    expect(sampleTab).toBeInTheDocument();
+    expect(customTab).toBeInTheDocument();
+
+    fireEvent.click(sampleTab);
+    const sampleArea = screen.getByTestId('input-sample') as HTMLTextAreaElement;
+    expect(sampleArea).toBeInTheDocument();
+    expect(sampleArea.value).toBe('s');
+
+    fireEvent.click(customTab);
+    const customArea = screen.getByTestId('input-custom') as HTMLTextAreaElement;
+    expect(customArea).toBeInTheDocument();
+    expect(customArea.value).toBe('c');
+  });
+
+  it('updates sample on change and triggers onSampleChange callback', () => {
+    const onSample = vi.fn();
+    render(<ConsoleIO sampleValue="" onSampleChange={onSample} data-testid="consoleio-root" />);
+
+    const sampleArea = screen.getByTestId('input-sample') as HTMLTextAreaElement;
+    fireEvent.change(sampleArea, { target: { value: 'new-s' } });
```

Comment:
Using `fireEvent.change` directly bypasses user typing behaviour and might not replicate real user interactions that trigger composition/input events.

**Sugerencia:**
```typescript
Use `const user = userEvent.setup(); await user.type(sampleArea, 'new-s');` (and assert on the callback) to better emulate user behaviour.
```